packages/components/src/CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+# Contributing components
+
+These guidelines document how to contribute to components in the [`@wordpress/components`](https://www.npmjs.com/package/@wordpress/components) npm package, or “WordPress Components”. This can be addition, modification, or deprecation.
+
+## Does it belong in the component library?
+
+A component library should include what’s shared across many products and omit what’s not. Questions to help determine if a component should be added:
+
+- Is it relevant to other products/plugins?
+- Is there overlap with any existing components?
+- How much will it cost to make and maintain?
+- Can you articulate how the component should be used?
+
+Here’s a flowchart that can help determine if a new component is necessary:
+
+[![New component flowchart](https://wordpress.org/gutenberg/files/2019/07/New_component_flowchart.png)](https://coggle.it/diagram/WtUSrld3uAYZHsn-/t/new-ui-component/992b38cbe685d897b4aec6d0dd93cc4b47c06e0d4484eeb0d7d9a47fb2c48d94)
+
+## First steps to contributing
+
+Once you’re ready to contribute to WordPress Components, start by opening a GitHub issue. Include a detailed description in which you:
+
+- Explain the rationale
+- Detail the intended behavior
+- Clarify whether it’s a variation of an existing component, or a new asset
+- Include mockups of any fidelity (optional)
+- Include any inspirations from other products (optional)
+
+This issue will be the staging ground for the contribution and an opportunity for the community to weigh in with any suggestions. Reviewers can also check if there is overlap with an existing component, or if there is a more appropriate location for the component.
+It’s encouraged to surface works-in-progress. If you’re not able to complete all of the parts yourself, someone in the community may be able to pick up where you left off.
+
+## Parts of a component contribution
```

Comment:
```suggestion
## Next steps
```

Once the team has discussed and approved the change, it's time to start implementing it.

Reply:
> Once the team has discussed and approved the change, it's time to start implementing it.

Did you mean to include this line in the suggestion? I like it.

Reply:
I sure did! I was trying to make the steps a bit more sequential & balanced, so "First steps" and only _then_ when that's all done should we move on to "Next steps."

We could also move the "you don't have to do it all yourself" bit up here as well, although I think it works well either before or after the list of steps.